### PR TITLE
MGDOBR-1181: Add service limits errors management when creating bridges and processors.

### PR DESCRIPTION
**Pull request by ppadti**
JIRA: https://issues.redhat.com/browse/MGDOBR-1181

Steps to test it (on chrome):

- Try to create a new bridge instance with name containing "quota-error".
- Submit the form by clicking on Create Smart Events Instance.
- alert will appear on the top of the form.
- Also try to create a new processor with name containing "quota-error".
- It will display a modal with error message.

### CODE REVIEW

**Review by jomarko — COMMENTED** [review]
Thank you for the PR. My only minor observation, is in naming (string constants) we introduce in this PR. I am not usre, if the values are:
- either comming form UX team ?
- or aligned with Backend?
- or ... ?

Why I am asking.
In `smart-events.json` we have: **deploy** a new instance
in `handlers.ts` we have: **Creation** of a bridge

What is the difference between `deploy` vs `create` ? Shouldn't we use just one of them? Similar question for `instance` vs `bridge`

Similar observation is in messages:
- "quotaExceededErrorDescription": "To deploy a new instance, delete an existing one first"
- "processorQuotaExceeded": "Your organization is out of quota. To create a new processor, delete the existing one"

Once we are deleting **an** existing one, second time **the** existing one. Shouldn we unify the message?

**Review by jomarko — COMMENTED** [review]
I prepared cypress tests for quota exceeded scenarios, please have a look @ppadti https://github.com/ppadti/sandbox-ui/pull/2

File: locales/en/smart-events.json [hunk 1]
```diff
@@ -113,7 +114,8 @@
     "selectRegion": "Select region",
     "smartEventInstance": "Smart Events Instance",
     "smartEventInstances": "Smart Event Instances",
-    "viewInformation": "View instance information"
+    "viewInformation": "View instance information",
+    "quotaExceededErrorDescription": "To deploy a new instance, delete an existing one first"
```
Comment (kelvah) [inline locales/en/smart-events.json:118]: Nitpicking, missing dot at the end.

```suggestion
    "quotaExceededErrorDescription": "To deploy a new instance, delete an existing one first."
```

File: locales/en/smart-events.json [hunk 2]
```diff
@@ -146,7 +148,8 @@
       "malformedTransformation": "The Transformation Template for the Processor is invalid.",
       "missingProcessorType": "Please select a processor type.",
       "parametersSectionGenericError": "Error while retrieving the information related to the parameters section for this Processor.",
-      "processorDetailsGenericError": "Error while retrieving the Processor details."
+      "processorDetailsGenericError": "Error while retrieving the Processor details.",
+      "processorQuotaExceeded": "Your organization is out of quota. To create a new processor, delete the existing one"
```
Comment (kelvah) [inline locales/en/smart-events.json:152]:
```suggestion
      "processorQuotaExceeded": "Your organization is out of quota. To create a new processor, delete an existing one first."
```

File: mocked-api/handlers.ts [hunk 3]
```diff
@@ -233,6 +233,21 @@ export const handlers = [
...
 
+    if (name.includes("quota-error")) {
+      return res(
+        ctx.status(402),
+        ctx.json({
+          kind: "ErrorsResponse",
+          items: [
+            {
+              ...error_quota_exceeded,
+              reason: `Creation of bridge was no successful due to exceeded quota'`,
```
Comment (kelvah) [inline mocked-api/handlers.ts:244]: You can use the same reason message the API returns. You can find it inside the `errors.ts` you updated, it's in the comment before the new error code.

Given it's a generic message for both SE instances and processors, you can also move it inside `error_quota_exceeded`.

**Review by kelvah — COMMENTED** [review]
Nice work @ppadti.
Only some minor adjustments in my comments.

**Comment by kelvah** [comment]
@jomarko I've proposed some changes in my review. They should cover the questions you asked.

**Comment by jomarko** [comment]
@kelvah thank you. I will wait for changes pushed by @ppadti

**Comment by jomarko** [comment]
Messages looks good to me @ppadti now. I am waiting for incorporating tests https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/250#pullrequestreview-1166739863 now.

Let me know, if you have any question for me.

Thank you

**Comment by ppadti** [comment]
> Messages looks good to me @ppadti now. I am waiting for incorporating tests [#250 (review)](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/250#pullrequestreview-1166739863) now.
> 
> Let me know, if you have any question for me.
> 
> Thank you

It's Done @jomarko

**Review by jomarko — APPROVED** [review]
Thank you for all the updates.

**Review by kelvah — APPROVED** [review]
Looking good. :+1: 
Thanks @ppadti.

**Review by vpellegrino — APPROVED** [review]
LGTM, thanks @ppadti

**Pull request merged**